pkg/executor/executor.go

```diff
@@ -116,7 +119,43 @@ func (executor *Executor) serveCreateFuncServices() {
 		req := <-executor.requestChan
 		fnMetadata := &req.function.ObjectMeta
 
+		var exactlyOneChan *multicast.Channel[*fscache.FuncSvc]
+		fnKey := fmt.Sprintf("%s:%s", req.function.UID, fnMetadata.ResourceVersion)
+
 		if req.function.Spec.InvokeStrategy.ExecutionStrategy.ExecutorType == fv1.ExecutorTypePoolmgr {
+
+			// This is a special case to ensure that if a Function spec has a concurrency of one
+			// we only ever spawn one of those pods.
+			// If we end up in this create function and see that another thread is already specializing
+			// we wait for it to broadcast the result instead of creating a new pod.
+			if req.function.Spec.Concurrency == 1 {
+				maybeChan, loaded := executor.exactlyOneChans.LoadOrStore(fnKey, multicast.New[*fscache.FuncSvc]())
+
+				exactlyOneChan = maybeChan.(*multicast.Channel[*fscache.FuncSvc])
+				listener := exactlyOneChan.Listen()
+
+				if loaded {
+					go func() {
+						ctx, cancel := context.WithTimeout(req.context, 10*time.Second)
```

Comment:
We discussed this IRL and we think this timeout should be the same as it is [below](https://github.com/gadget-inc/fission/pull/10/files#diff-b41713213596a17bc034a9cc9d33e2f26ee345da66c23ca56154be85c5b44c44R160-R172).